controller/server/main.go

```diff
@@ -356,6 +357,60 @@ func (s *server) ShowTopology(ctx context.Context, req *cpb.ShowTopologyRequest)
 	return resp, nil
 }
 
+func (s *server) PushConfig(ctx context.Context, req *cpb.PushConfigRequest) (*cpb.PushConfigResponse, error) {
+	log.Infof("Received PushConfig request: %v", req)
+	s.muTopo.Lock()
```

Comment:
do we really want a  global lock for this operation?

Reply:
I decided to use the lock here to prevent a request to a different method from changing a topo while this method relies on that topos state and existence.

This would be more efficient if we had one lock per topology (for edits like these) and only use this lock for updating the topology map